d2renderers/d2svg/d2svg.go

```diff
@@ -746,6 +746,128 @@ func render3dRect(targetShape d2target.Shape) string {
 	return borderMask + mainShapeRendered + renderedSides + renderedBorder
 }
 
+func render3dHexagon(targetShape d2target.Shape) string {
+	moveTo := func(p d2target.Point) string {
+		return fmt.Sprintf("M%d,%d", p.X+targetShape.Pos.X, p.Y+targetShape.Pos.Y)
+	}
+	lineTo := func(p d2target.Point) string {
+		return fmt.Sprintf("L%d,%d", p.X+targetShape.Pos.X, p.Y+targetShape.Pos.Y)
+	}
+	scale := func(n int, f float64) int {
+		return int(float64(n) * f)
+	}
+	halfYFactor := 43.6 / 87.3
+
+	// draw border all in one path to prevent overlapping sections
+	var borderSegments []string
+	borderSegments = append(borderSegments,
+		moveTo(d2target.Point{X: scale(targetShape.Width, 0.25), Y: 0}),
+	)
+	Y_OFFSET := d2target.THREE_DEE_OFFSET / 2
+	for _, v := range []d2target.Point{
```

Comment:
maybe leave a comment on which part of the 3d this is drawing. e.g. the top side vs the right side

Reply:
got it. I've added the comments